packages/jest-types/__typetests__/jest.test.ts

```diff
@@ -7,89 +7,292 @@
 expectType<typeof jest>(jest.mock('moduleName', jest.fn(), {}));
 expectType<typeof jest>(jest.mock('moduleName', jest.fn(), {virtual: true}));
-expectType<typeof jest>(jest.resetModules());
-expectType<typeof jest>(jest.isolateModules(() => {}));
-expectType<typeof jest>(jest.retryTimes(3));
-expectType<Mock<Promise<string>, []>>(
-  jest
-    .fn(() => Promise.resolve('string value'))
-    .mockResolvedValueOnce('A string, not a Promise'),
+expectError(jest.mock());
+
+expectType<typeof jest>(jest.unstable_mockModule('moduleName', jest.fn()));
+expectType<typeof jest>(
+  jest.unstable_mockModule('moduleName', () => Promise.resolve(jest.fn())),
 );
-expectType<Mock<Promise<string>, []>>(
-  jest
-    .fn(() => Promise.resolve('string value'))
-    .mockResolvedValue('A string, not a Promise'),
+expectType<typeof jest>(jest.unstable_mockModule('moduleName', jest.fn(), {}));
+expectType<typeof jest>(
+  jest.unstable_mockModule('moduleName', () => Promise.resolve(jest.fn()), {}),
 );
-expectType<Mock<Promise<string>, []>>(
-  jest
-    .fn(() => Promise.resolve('string value'))
-    .mockRejectedValueOnce(new Error('An error, not a string')),
+expectType<typeof jest>(
+  jest.unstable_mockModule('moduleName', jest.fn(), {virtual: true}),
 );
-expectType<Mock<Promise<string>, []>>(
-  jest
-    .fn(() => Promise.resolve('string value'))
-    .mockRejectedValue(new Error('An error, not a string')),
+expectType<typeof jest>(
+  jest.unstable_mockModule('moduleName', () => Promise.resolve(jest.fn()), {
+    virtual: true,
+  }),
 );
 
-expectType<void>(jest.runAllImmediates());
-expectType<void>(jest.runAllTicks());
-expectType<void>(jest.runAllTimers());
-expectType<void>(jest.runOnlyPendingTimers());
-expectType<void>(jest.advanceTimersByTime(9001));
+expectType<unknown>(jest.requireActual('./pathToModule'));
+expectError(jest.requireActual());
+
+expectType<unknown>(jest.requireMock('./pathToModule'));
+expectError(jest.requireMock());
+
+expectType<typeof jest>(jest.resetModules());
+expectError(jest.resetModules('moduleName'));
 
-expectType<typeof jest>(jest.setMock('moduleName', {}));
-expectType<typeof jest>(jest.setMock('moduleName', {}));
 expectType<typeof jest>(jest.setMock('moduleName', {a: 'b'}));
-expectType<typeof jest>(jest.setTimeout(9001));
+expectError(jest.setMock('moduleName'));
+
 expectType<typeof jest>(jest.unmock('moduleName'));
-expectType<typeof jest>(jest.useFakeTimers());
-expectType<typeof jest>(jest.useRealTimers());
+expectError(jest.unmock());
+
+// Mock Functions
+
+expectType<typeof jest>(jest.clearAllMocks());
+expectError(jest.clearAllMocks('moduleName'));
+
+expectType<boolean>(jest.isMockFunction(() => {}));
+expectError(jest.isMockFunction());
+
+const maybeMock = (a: string, b: number) => true;
+
+if (jest.isMockFunction(maybeMock)) {
+  expectType<Mock<boolean, [a: string, b: number]>>(maybeMock);
+
+  maybeMock.mockReturnValueOnce(false);
+  expectError(maybeMock.mockReturnValueOnce(123));
```

Comment:
TS is wrapping the function in `Mock` type here. I think that’s fine, this branch will never run anyways.